From e36f677c4ab315a4196aeb261a867f2982117054 Mon Sep 17 00:00:00 2001
From: Jonathan Maple <jmaple@ciq.com>
Date: Tue, 24 Feb 2026 16:48:08 -0500
Subject: [PATCH] kdb: Remove currently unused kdbtab_t->cmd_flags

jira VULN-1501
cve-pre CVE-2022-21499
commit-author Anton Vorontsov <anton.vorontsov@linaro.org>
commit a2e5d188aad31f7177cbd6d9ddaf8cc9aa4affe0

The struct member is never used in the code, so we can remove it.

We will introduce real flags soon by renaming cmd_repeat to cmd_flags.

	Signed-off-by: Anton Vorontsov <anton.vorontsov@linaro.org>
	Signed-off-by: John Stultz <john.stultz@linaro.org>
	Signed-off-by: Daniel Thompson <daniel.thompson@linaro.org>
	Cc: Jason Wessel <jason.wessel@windriver.com>
	Signed-off-by: Jason Wessel <jason.wessel@windriver.com>
(cherry picked from commit a2e5d188aad31f7177cbd6d9ddaf8cc9aa4affe0)
	Signed-off-by: Jonathan Maple <jmaple@ciq.com>
---
 kernel/debug/kdb/kdb_main.c    | 1 -
 kernel/debug/kdb/kdb_private.h | 1 -
 2 files changed, 2 deletions(-)

diff --git a/kernel/debug/kdb/kdb_main.c b/kernel/debug/kdb/kdb_main.c
index 0b097c8a1e50..84254a5939e6 100644
--- a/kernel/debug/kdb/kdb_main.c
+++ b/kernel/debug/kdb/kdb_main.c
@@ -2694,7 +2694,6 @@ int kdb_register_repeat(char *cmd,
 	kp->cmd_func   = func;
 	kp->cmd_usage  = usage;
 	kp->cmd_help   = help;
-	kp->cmd_flags  = 0;
 	kp->cmd_minlen = minlen;
 	kp->cmd_repeat = repeat;
 
diff --git a/kernel/debug/kdb/kdb_private.h b/kernel/debug/kdb/kdb_private.h
index 7afd3c8c41d5..c4c46c7b26fd 100644
--- a/kernel/debug/kdb/kdb_private.h
+++ b/kernel/debug/kdb/kdb_private.h
@@ -172,7 +172,6 @@ typedef struct _kdbtab {
 	kdb_func_t cmd_func;		/* Function to execute command */
 	char    *cmd_usage;		/* Usage String for this command */
 	char    *cmd_help;		/* Help message for this command */
-	short    cmd_flags;		/* Parsing flags */
 	short    cmd_minlen;		/* Minimum legal # command
 					 * chars required */
 	kdb_repeat_t cmd_repeat;	/* Does command auto repeat on enter? */
-- 
2.47.3

